app/src/test/java/io/apicurio/registry/rest/v2/GroupsResourceTest.java

```diff
@@ -295,6 +297,87 @@ public void testCreateArtifact() throws Exception {
                 .body("id", equalTo("testCreateArtifact/EmptyAPI-customVersion"))
                 .body("type", equalTo(ArtifactType.OPENAPI.name()));
 
+        // Create OpenAPI artifact - provide a custom name
+        String customName = "CUSTOM NAME";
+        given()
+            .when()
+                .contentType(CT_JSON + "; artifactType=OPENAPI")
+                .pathParam("groupId", GROUP)
+                .header("X-Registry-ArtifactId", "testCreateArtifact/EmptyAPI-customName")
+                .header("X-Registry-Name", customName)
+                .body(artifactContent)
+                .post("/registry/v2/groups/{groupId}/artifacts")
+            .then()
+                .statusCode(200)
+                .body("groupId", equalTo(GROUP))
+                .body("name", equalTo(customName))
+                .body("id", equalTo("testCreateArtifact/EmptyAPI-customName"))
+                .body("type", equalTo(ArtifactType.OPENAPI.name()));
+
+        // Create OpenAPI artifact - provide a custom No-ASCII name
+        String customNoASCIIName = "CUSTOM NAME with NO-ASCII char č";
+        given()
+            .when()
+                .contentType(CT_JSON + "; artifactType=OPENAPI")
+                .pathParam("groupId", GROUP)
+                .header("X-Registry-ArtifactId", "testCreateArtifact/EmptyAPI-customNameEncoded")
+                .header("X-Registry-Name-Encoded", Base64.encode(customNoASCIIName.getBytes(StandardCharsets.UTF_8)))
+                .body(artifactContent)
+                .post("/registry/v2/groups/{groupId}/artifacts")
+            .then()
+                .statusCode(200)
+                .body("groupId", equalTo(GROUP))
+                .body("name", equalTo(customNoASCIIName))
+                .body("id", equalTo("testCreateArtifact/EmptyAPI-customNameEncoded"))
+                .body("type", equalTo(ArtifactType.OPENAPI.name()));
+
+        // Create OpenAPI artifact - provide a custom description
+        String customDescription = "CUSTOM DESCRIPTION";
+        given()
+            .when()
+                .contentType(CT_JSON + "; artifactType=OPENAPI")
+                .pathParam("groupId", GROUP)
+                .header("X-Registry-ArtifactId", "testCreateArtifact/EmptyAPI-customDescription")
+                .header("X-Registry-Description", customDescription)
+                .body(artifactContent)
+                .post("/registry/v2/groups/{groupId}/artifacts")
+            .then()
+                .statusCode(200)
+                .body("groupId", equalTo(GROUP))
+                .body("description", equalTo(customDescription))
+                .body("id", equalTo("testCreateArtifact/EmptyAPI-customDescription"))
+                .body("type", equalTo(ArtifactType.OPENAPI.name()));
+
+        // Create OpenAPI artifact - provide a custom No-ASCII description
+        String customNoASCIIDescription = "CUSTOM DESCRIPTION with NO-ASCII char č";
```

Comment:
This will be interesting.  How does our toolchain handle unicode characters?  :)